tests/target/issue-2869.rs

```diff
@@ -0,0 +1,36 @@
+// rustfmt-struct_field_align_threshold: 50
+
+#[derive(Serialize, Deserialize, Debug)]
+#[serde(rename_all = "PascalCase")]
+struct AuditLog1 {
+    creation_time:                     String,
+    id:                                String,
+    operation:                         String,
+    organization_id:                   String,
+    record_type:                       u32,
+    result_status:                     Option<String>,
+    #[serde(rename = "ClientIP")]
+    client_ip:                         Option<IpAddr>,
```

Comment:
I think we should align items in a different group if there is an attribute in between:

```rust
#[derive(Serialize, Deserialize, Debug)]
#[serde(rename_all = "PascalCase")]
struct AuditLog1 {
    creation_time:   String,
    id:              String,
    operation:       String,
    organization_id: String,
    record_type:     u32,
    result_status:   Option<String>,
    #[serde(rename = "ClientIP")]
    client_ip:                         Option<IpAddr>,
    object_id:                         String,
    actor:                             Option<Vec<IDType>>,
    actor_context_id:                  Option<String>,
    actor_ip_address:                  Option<IpAddr>,
    azure_active_directory_event_type: Option<u8>,
}
```

Empty lines and attributes should define alignment group as well.

Reply:
In the original the types were aligned which is why I tried to reproduce it

Reply:
from https://github.com/rust-lang/rustfmt/pull/3513#issuecomment-500213375 groups are now defined by blank lines.